src/main/java/org/eclipse/yasson/internal/ConstructorPropertiesAnnotationIntrospector.java

```diff
@@ -43,6 +44,11 @@ public JsonbCreator getCreator(Constructor<?>[] constructors) {
             if (!(properties instanceof String[])) {
                 continue;
             }
+            if (!Modifier.isPublic(constructor.getModifiers())) {
+                String declaringClass = constructor.getDeclaringClass().getName();
+                LOG.finest(Messages.getMessage(MessageKeys.INACCESSIBLE_CONSTRUCTOR_PROPERTIES_CREATOR, declaringClass, Arrays.toString((String[]) properties)));
```

Comment:
Had a second thought on this one -- since this situation represents a static error in the code I think we should blow up with the message you've written. I see no reason to tolerate an extra annotation which will only ever be ignored by JSON-B

Reply:
If "we should blow up" means that we should throw an exception, then this would not solve https://github.com/payara/Payara/issues/4003.

I would agree with you, when it comes to value objects, that only have this one annotated but inaccessible constructor, and no other way to construct them. This also meets the spec.

BUT :-) If there is no explicitly annotated creator, then the "normal" no-arg constructor is used to create an instance (see `ObjectDeserializer.getInstance()`). This should work, regardless of any additional constructor, that is annotated with @ConstructorProperties.

Value Objects with:
- public no-arg constructor
- getter/setter or field access
- one additional, non-public constructor annotated with @ConstructorProperties

will fail to get deserialized, only because of the @ConstructorProperties-support, that came with 1.0.4. (ok, that's on me :-))

One way to solve this is to log an inaccessible @ConstructorProperties annotated constructor and ignore it. There may be a no-arg constructor later and everything is fine. There may be no way to create an instance and the deserialization would fail a bit later than it should (not that nice indeed).

The other way would be to throw an exception, but only in the case, when there is no useable other way to create an instance. This would meet the "fail fast" principle, but will lead to do the "normal" detection earlier during getCreator() additional to `ObjectDeserializer.getInstance()`. This could get a bit complicated and eventually messy.

One interesting thing last but not least:
I added
"testNoArgConstructorShouldBePreferredOverUnusableJsonbAnnotatedProtectedConstructor" to "AnnotationIntrospectorTest". This test case is currently ignored, since it does not work as i'd expect. It shows, that the problem also applies to @JsonbCreator-annotated, inaccessible constructors with an additional valid no-arg constructor. The specification does not forbid the current behavior, as far as i've seen. 

Both of these special cases shouldn't come up that often. But since there are generators and tools like lombok, value-objects may not always end up as they would in a hand written way.

Reply:
I suppose we can just log a message and move on as you suggest, but since this is a FINEST log level, lets not use a translated message from the yasson-messages resource file